src/hooks/useResponsive.js

```diff
@@ -0,0 +1,30 @@
+import { useContext, useEffect } from 'react'
+import { ResponsiveContext } from 'grommet'
+
+/*
+  small = a phone <= 500
+  medium = a tablet <= 1024 
+  large = a computer >= 1025
+*/
+
+export const useResponsive = () => {
+  const viewport = useContext(ResponsiveContext)
+
+  useEffect(() => {}, [viewport])
+
+  const setResponsive = (small, medium, large) => {
```

Comment:
```suggestion
  const getResponsive = (small, medium, large) => {
```

I know this is probably an annoying change but setters should set something. You can make this change later but I think the semantics of getting and setting are important. Getters retrieve a value, setters modify a state or data structure.

Reply:
This helper sets css values in JSX(for UI) based on different viewports, so semantically it makes sense. Yes, let's keep it as is for now, thank you!

Reply:
Yeah let me explain this a little bit better...

```js
// example hook code
const data = {}
const setAttribute = (newVal, attribute) => {
  data[attribute] = newVal
  return data
}
const getAttribute = (attrbute) => data[attribute]

// example use
<Component attribute={getAttribute("attribute")} />
```

Even though the `getAttribute` function is returning a value that then sets a value in a component, that does not make it a "setter". As you can see in the example, a setter modifies something during its invocation, a getter returns a value after invocation. 

This is function should be named `getResonsive` since it returns a value. It does not modify any data inside the function.